From 94995e9c1e1ffc3eb44c06d23a30a2ac8cd2455d Mon Sep 17 00:00:00 2001
From: Carlos Garnacho <carlosg@gnome.org>
Date: Tue, 30 Apr 2019 14:51:47 +0200
Subject: [PATCH] st: Reload icon on style changes only if necessary

The parameters that may affect the icon on ::style-changed are more size
related than visual (we listen to icon theme changes for the latter). It
makes sense to just update the icon if the size came out different.

https://gitlab.gnome.org/GNOME/gnome-shell/merge_requests/524
---
 src/st/st-icon.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/st/st-icon.c b/src/st/st-icon.c
index 04d2d1d3c..8b7015281 100644
--- a/src/st/st-icon.c
+++ b/src/st/st-icon.c
@@ -206,8 +206,8 @@ st_icon_style_changed (StWidget *widget)
     }
 
   priv->theme_icon_size = (int)(0.5 + st_theme_node_get_length (theme_node, "icon-size"));
-  st_icon_update_icon_size (self);
-  st_icon_update (self);
+  if (st_icon_update_icon_size (self))
+    st_icon_update (self);
 }
 
 static void
-- 
2.18.1

